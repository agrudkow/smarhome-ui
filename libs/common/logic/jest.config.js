module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
