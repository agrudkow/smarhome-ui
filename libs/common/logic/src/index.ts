export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
