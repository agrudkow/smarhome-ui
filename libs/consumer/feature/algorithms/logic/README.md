# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
