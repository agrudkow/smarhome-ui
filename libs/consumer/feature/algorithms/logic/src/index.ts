export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
