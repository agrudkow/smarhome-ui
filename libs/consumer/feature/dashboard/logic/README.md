# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
