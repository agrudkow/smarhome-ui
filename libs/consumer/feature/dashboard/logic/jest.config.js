module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
