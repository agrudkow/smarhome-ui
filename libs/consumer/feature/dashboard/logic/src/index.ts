export * from './lib/use-billings';
