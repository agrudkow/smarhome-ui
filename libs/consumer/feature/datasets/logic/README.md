# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
