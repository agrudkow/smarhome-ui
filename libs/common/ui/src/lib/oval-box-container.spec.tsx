import React from 'react';
import { customRender } from '@smarthome/common/logic';
import OvalBoxContainer from './oval-box-container';

describe(' OvalBoxContainer', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(<OvalBoxContainer />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(<OvalBoxContainer />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should set height', () => {
    const height = 200;
    const { container } = customRender(<OvalBoxContainer height={height} />);
    expect(container.firstChild).toHaveStyleRule('height', `${height}px`);
  });

  it('should set width', () => {
    const width = 200;
    const { container } = customRender(<OvalBoxContainer width={width} />);
    expect(container.firstChild).toHaveStyleRule('width', `${width}px`);
  });

  it('should set backgroundColor', () => {
    const backgroundColor = 'red';
    const { container } = customRender(
      <OvalBoxContainer backgroundColor={backgroundColor} />
    );
    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      backgroundColor
    );
  });

  it('should set opacity', () => {
    const opacity = 0.4;
    const { container } = customRender(<OvalBoxContainer opacity={opacity} />);
    expect(container.firstChild).toHaveStyleRule('opacity', `${opacity}`);
  });

  it('should set boxShadow', () => {
    const boxShadow = true;
    const { container } = customRender(
      <OvalBoxContainer boxShadow={boxShadow} />
    );
    expect(container.firstChild).toHaveStyleRule(
      'box-shadow',
      '0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)'
    );
  });
});
