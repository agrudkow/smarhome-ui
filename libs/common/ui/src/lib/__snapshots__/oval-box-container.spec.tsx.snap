// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` OvalBoxContainer should match snapshot 1`] = `
.c0 {
  padding: 5px 15px;
  margin: 15px 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  width: auto;
  max-height: auto;
  max-width: auto;
  background-color: #fff;
  opacity: 1;
  border-radius: 6px;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

@media (max-width:769px) {
  .c0 {
    margin: 5px 0;
  }
}

@media (max-width:480px) {
  .c0 {
    padding: 5px 5px;
  }
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;
